libenkf/src/ert_run_context.c

```diff
@@ -146,12 +191,39 @@ ert_run_context_type * ert_run_context_alloc_SMOOTHER_RUN(enkf_fs_type * simulat
   return context;
 }
 
+ert_run_context_type * ert_run_context_alloc(run_mode_type run_mode,
+                                             init_mode_type init_mode,
+                                             enkf_fs_type * init_fs,
+                                             enkf_fs_type * simulate_fs ,
+                                             enkf_fs_type * target_update_fs ,
+                                             bool_vector_type * iactive ,
+                                             path_fmt_type * runpath_fmt ,
+                                             subst_list_type * subst_list ,
+                                             int iter) {
+  if (run_mode == SMOOTHER_UPDATE)
```

Comment:
Please consider using a switch for handling various enum values. The compiler will then help us keeping track of when a specific value is not handled.